### PR TITLE
Change init logger to include timestamp for debug level

### LINKED ISSUE #21100: `tsh -d` should include timestamps with logs
If you run `tsh -d` it will output logs to the console but not include timestamps which sometimes makes debugging difficult:

```
➜  ~ tsh -d login --proxy=root.gravitational.io:3080 --auth=local --user=alice
INFO [CLIENT]    no host login given. defaulting to r0mant client/api.go:954
INFO [CLIENT]    [KEY AGENT] Connected to the system agent: "/private/tmp/com.apple.launchd.TmAlgBvMpv/Listeners" client/api.go:3884
DEBU [CLIENT]    attempting to use loopback pool for local proxy addr: root.gravitational.io:3080 client/api.go:3842
DEBU [CLIENT]    reading self-signed certs from: /var/lib/teleport/webproxy_cert.pem client/api.go:3850
DEBU [CLIENT]    could not open any path in: /var/lib/teleport/webproxy_cert.pem client/api.go:3854
DEBU             Attempting GET root.gravitational.io:3080/webapi/ping/local webclient/webclient.go:129
DEBU [CLIENT]    Attempting to login with a new RSA private key. client/api.go:3264
...
```

Let's make sure that timestamps are included in the debug log output by default.

### PR DESCRIPTION
**Pull request by EdwardDowling**
closes #21100
This will also change the behavior for teleport, tbot and tctl. 
I can change Pr to affect these changes only for tsh but it seems like the same arguments for adding the timestamp to tsh would apply for those as well.

### CODE REVIEW

**Comment by gzdunek** [comment]
I think we would like to disable this new behavior for Connect because it makes our log file look like this in development mode:
```
[13-02-23 18:20:39] 2023-02-13T18:20:39+01:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-02-14 03:31:15 +0000 UTC". client/client_store.go:89
```
(`tsh` stdout is passed to JS were we add timestamps)

We have already [customized the logging for non-debug mode](https://github.com/gravitational/teleport/blob/a93aaa7e6e0c535247c35fffecf80ae00dc7788b/tool/tsh/daemon.go#L39), so perhaps it should be done similarly for debug.

**Comment by EdwardDowling** [comment]
> 

@gzdunek I'm not entirely sure what you mean here.
Is it only in development mode that the stdout is passed to JS?
And what exactly do you mean by diabling this just for Connect?

**Comment by gzdunek** [comment]
Ah, I didn't mention that Connects runs tsh daemon in debug mode while in development.
`tsh` stdout is passed to JS (Electron) in both dev and prod mode, where its is processed (mainly by adding timestamps). We have multiple sources of logs in the app, which are processed in the same way. 

After your change, the timestamp is duplicated when Connect is in dev mode (one timestamp comes from our JS code, and one from `tsh`).

By disabling this new behavior I mean that `tsh daemon` can override default `tsh` logging when it was started in debug mode:
```go
if cf.Debug {
	utils.InitLogger(utils.LoggingForDaemon, logrus.DebugLevel)
} else {
	utils.InitLogger(utils.LoggingForDaemon, logrus.InfoLevel)
}
```
(in the code I linked)

File: tool/tsh/daemon.go [hunk 1]
```diff
@@ -35,7 +35,9 @@ func onDaemonStart(cf *CLIConf) error {
 
 	// Use info-level daemon-grade logging for the daemon running in non-debug mode.
 	// tsh already sets debug-level CLI-grade logging when running in debug mode.
```
Comment (gzdunek) [inline tool/tsh/daemon.go:37]: We can remove this comment now.

**Review by gzdunek — APPROVED** [review]

**Comment by EdwardDowling** [comment]
@zmb3 can you take a quick look at this when you get a chance?

**Review by zmb3 — APPROVED** [review]

**Pull request merged**